orca/orca.go

```diff
@@ -68,29 +70,42 @@ func CallMetricsServerOption() grpc.ServerOption {
 
 func unaryInt(ctx context.Context, req interface{}, _ *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
 	recorder := newMetricRecorder()
-	r := &CallMetricRecorder{MetricSetter: recorder}
-	ctxWithRecorder := setCallMetricRecorder(ctx, r)
+	ctxWithRecorder := newContextWithMetricSetter(ctx, recorder)
 	resp, err := handler(ctxWithRecorder, req)
-	if err2 := recorder.setTrailerMetadata(ctx); err2 != nil {
-		logger.Warning(err2)
-	}
+	setTrailerMetadata(ctx, recorder.toLoadReportProto())
```

Comment:
So this might be too obfuscated, but consider:

```go
defer setTrailerMetadata(ctx, recorder) // note we can't pass the proto since it would be evaluated _now_
return handler(ctxWithRecorder, req)
```

Reply:
Done.